libsolidity/interface/SMTSolverCLI.h

```diff
@@ -0,0 +1,50 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <libsolidity/interface/ReadFile.h>
+
+#include <boost/filesystem.hpp>
+
+namespace solidity::frontend
+{
+
+class SMTSolverCLI
```

Comment:
How about this? I was a bit confused about the CLI in the name until I realized you really meant mostly the 'command' part rather than the 'interface' part :)

```suggestion
class SMTSolverCommand
```

Also, currently it's hard-coded for Eldarica so I'd rename it to `EldaricaCommand`. Or you could make it actually more generic and accept `modelChecker.solvers` as input and decide based on that (i.e. just do nothing for other solvers but at least keep that decision here). And then I'd hard-code the solver command here (rather than in the CLI).

Reply:
Changed the name.
This class is not hardcoded to Eldarica though, that's a constructor argument.

Reply:
It actually is, and the argument seems unused, though I suspect it's just a bug.

But ignoring that, is it usable with any other solver? It gives the binary arguments like `-ssol` and `-scex`. I'm assuming that's specific to Eldarica?

Reply:
Yea I removed those in this version actually, because we don't use them anyway in the current version. It should be usabled with any solver.